core/shared/src/main/scala/zio/stm/TMap.scala

```diff
@@ -233,6 +233,43 @@ final class TMap[K, V] private (
       TExit.unit
     })
 
+  /**
+   * Removes bindings matching predicate and returns the removed entries.
+   */
+  def removeIfReturning(p: (K, V) => Boolean): USTM[Chunk[(K, V)]] =
```

Comment:
My cup of tea:
- If we go with two combinators we should probably follow the approach from `foreach` case, i.e. `foreach_` was renamed to `foreachDiscard`. Unfortunately, using it here would mean we have to postpone this change for 2.x.
- Regardless of the approach, please provide the same alternative for `retainIf` :).

Reply:
I also wrote the `retainIfReturning` (which uses `removeIfReturning` under the hood). I don't really have a preference with regards to the naming so I'm happy to apply any suggestion from your side

Reply:
Yeah I think the `Returning` suffix is confusing and it is not used anywhere in our code base. The `Discard` suffix is not ideal in a way because it makes the more common use case where you don't care about the values longer but it is at least consistent. Let's retarget to the `series/2.x` branch and do that.

Reply:
@jeroenr please rename the combinators as follows:

old name | new name
-|-
`removeIf` | `removeIfDiscard`
`removeIfReturning` | `removeIf`
`retainIf` | `retainIfDiscard`
`retainIfReturning` | `retainIf`

As Adam pointed out, these are unfortunate, but if we find better names in the meantime, the change is "one `sed` away" :).

Reply:
I applied the suggested renaming and also introduced the same in `TSet` to keep APIs in sync. PTAL :)